UI/src/model/selectedLaneModelling.js

```diff
@@ -0,0 +1,578 @@
+(function(root) {
+  root.SelectedLaneModelling = function(backend, collection, typeId, singleElementEventCategory, multiElementEventCategory, isSeparableAssetType) {
+    var lanesFetched = [];
+    var selection = [];
+    var selectedRoadlink = null;
+    var assetsToBeExpired = [];
+    var assetsToBeRemoved = [];
+    var self = this;
+    var dirty = false;
+    var linksSelected = null;
+    var multipleSelected;
+    var currentLane;
+
+    var initial_road_number;
+    var initial_road_part_number;
+    var initial_distance;
+    var end_road_part_number;
+    var end_distance;
+    var track;
+
+    this.getLane = function (laneNumber, marker) {
+        return _.find(selection, function (lane){
+          return (_.isEmpty(marker) || lane.marker == marker) && _.find(lane.properties, function (property) {
+            return property.publicId === "lane_code" && _.head(property.values).value == laneNumber;
+          });
+        });
+    };
+
+    this.getCurrentLaneNumber = function() {
+      if(!_.isUndefined(currentLane)) {
+        return _.head(_.find(currentLane.properties, {'publicId': 'lane_code'}).values).value;
+      }
+    };
+
+    this.getCurrentLane = function () { return currentLane; };
+
+    this.setCurrentLane = function (lane) { currentLane = self.getLane(lane); };
+
+    var reorganizeLanes = function (laneNumber) {
+      var lanesToUpdate = _.map(selection, function (lane){
+        var foundValidProperty =  _.find(lane.properties, function (property) {
+          if(_.isEmpty(property.values))
+            return false;
+
+          var value = _.head(property.values).value;
+          return property.publicId === "lane_code" && value > laneNumber && ((value % 2 !== 0 && laneNumber % 2 !== 0) || (value % 2 === 0 && laneNumber % 2 === 0));
+        });
+
+        if(_.isUndefined(foundValidProperty)){
+          return undefined;
+        }else{
+          return lane;
+        }
+      });
+
+      var listLanesIndexes = _.filter(_.map(lanesToUpdate,function (laneToUpdate) {
+        return _.findIndex(selection, function (lane) {
+          return lane == laneToUpdate;
+        });
+      }),function (index) {
+        return index != "-1";
+      });
+
+      if (!_.isEmpty(listLanesIndexes))
+        _.forEach(listLanesIndexes, function (number) {
+          var propertyIndex =  _.findIndex(selection[number].properties, function (property) {
+            return property.publicId === "lane_code";
+          });
+          selection[number].properties[propertyIndex].values[0].value = parseInt(selection[number].properties[propertyIndex].values[0].value) - 2;
+        });
+    };
+
+    var giveSplitMarkers = function(lanes){
+      var numberOfLanesByLaneCode = _.countBy(lanes, function (lane) {
+        return _.head(_.find(lane.properties, function (property) {
+          return property.publicId === "lane_code";
+        }).values).value;
+      });
+
```

Comment:
This code
```js
 _.head(_.find(lane.properties, function (property) {
          return property.publicId === "lane_code";
        }).values).value;
```
Is used multiple times. Why not a function for this?

Reply:
Done